packages/strapi-design-system/src/InputJSON/InputJSON.js

```diff
@@ -0,0 +1,127 @@
+import React, { useEffect, useRef, useCallback } from 'react';
+import PropTypes from 'prop-types';
+import styled from 'styled-components';
+import { Field, FieldLabel, FieldError } from '../Field';
+import { Box } from '../Box';
+import { Stack } from '../Stack';
+import { addMarks, filterMarks, lineHighlightMark } from './decorationExtension';
+import * as jsonlint from './utils/jsonLint';
+import { JsonComponent } from './JsonComponent';
+
+const WAIT = 500;
+
+const StyledBox = styled(Box)`
+  border-radius: ${({ theme }) => theme.borderRadius};
+  outline: 1px solid ${({ theme, error }) => (error ? theme.colors.danger600 : 'transparent')};
+`;
+
+export const InputJSON = ({ id, label, value, error, theme, onChange, editable }) => {
+  const editorState = useRef(null);
+  const editorView = useRef(null);
+  const timerRef = useRef();
+
+  const getContentAtLine = (line) => {
+    return editorState.current?.doc?.line(line);
+  };
+
+  const markSelection = useCallback(({ message }) => {
+    let line = parseInt(message.split(':')[0].split('line ')[1], 10);
+    const { text, to: lineEnd } = getContentAtLine(line);
+    const lineStart = lineEnd - text.trimStart().length;
+
+    if (lineEnd > lineStart)
+      editorView.current.dispatch({
+        effects: addMarks.of([lineHighlightMark.range(lineStart, lineEnd)]),
+      });
+  }, []);
+
+  const clearErrorHighlight = () => {
+    const docEnd = editorState.current?.doc?.length || 0;
+    editorView.current.dispatch({
+      effects: filterMarks.of((from, to) => to <= 0 || from >= docEnd),
+    });
+  };
+
+  const validateJSON = useCallback(
+    ({ value, isOnChangeCallback }) => {
+      try {
+        const formattedData = jsonlint.parse(value);
+
+        // callback to update json in parent
+        if (isOnChangeCallback) onChange(formattedData);
+      } catch (error) {
+        markSelection(error);
+      }
+    },
+    [markSelection, onChange],
+  );
+
+  const handleValidateJSON = useCallback(
+    ({ value, isOnChangeCallback }) => {
+      if (timerRef.current) {
+        clearTimeout(timerRef.current);
+      }
+
+      if (value) {
+        timerRef.current = setTimeout(() => validateJSON({ value, isOnChangeCallback }), WAIT);
```

Comment:
this serves 2 purpose :
1. It's mainly required for onChange handler, to avoid validating value for each and every input change. 
2. Also on load, if there is a wrong json string, we need validate it and highlight in the editor. So to get the editor state, we have createEditor callback which gets fired after useEffect. and so delay helps to get the editor state in the meantime.

Reply:
@joshuaellis can you please specify how you would do debounce in other way?

Reply:
i'd look at using a lib – perhaps https://github.com/biw/tiny-throttle which seems relatively harmless on bundle size :)

Reply:
Thanks I updated it. but I find [this](https://www.npmjs.com/package/throttle-debounce) more stable and popular except smaller bundle size than tiny-throttle. 
https://www.npmjs.com/package/throttle-debounce